packages/expo-sensors/ios/EXSensors/EXSensorsManager.m

```diff
@@ -286,6 +289,38 @@ - (BOOL)isMagnetometerUncalibratedAvailable
   return [[self manager] isMagnetometerAvailable];
 }
 
+
+- (void)proximitySensorStateChange:(NSNotification *)notification
+{
+    BOOL proximityState = [[UIDevice currentDevice] proximityState];
+    for (void (^handler)(NSDictionary *) in strongSelf.proximityHandlers.allValues) {
+      handler(@{@"proximityState": @(proximityState)});
+    }
+}
+
+- (void)sensorModuleDidSubscribeForProximityUpdates:(id)scopedSensorModule
+                                        withHandler:(void (^)(NSDictionary *event))handlerBlock
+{
+  if ([self isProximityAvailable]) {
+    _proximityHandlers[scopedSensorModule] = handlerBlock;
+  }
+  [[UIDevice currentDevice] setProximityMonitoringEnabled:YES];
+  [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(proximitySensorStateChange:) name:@"UIDeviceProximityStateDidChangeNotification" object:nil];
+}
+
+- (void)sensorModuleDidUnsubscribeForProximityUpdates:(id)scopedSensorModule
+{
+  [_proximityHandlers removeObjectForKey:scopedSensorModule];
+  if (_proximityHandlers.count == 0) {
+	[[UIDevice currentDevice] setProximityMonitoringEnabled:NO];
+  }
+}
+
+- (BOOL)isProximityAvailable
+{
+  return YES;
```

Comment:
Not always true: https://developer.apple.com/documentation/uikit/uidevice/1620017-proximitymonitoringenabled need to attempt to enable it and determine if the value is still false.